staging/src/k8s.io/apiserver/pkg/server/config.go

```diff
@@ -911,7 +911,10 @@ func (c completedConfig) New(name string, delegationTarget DelegationTarget) (*G
 	if s.isPostStartHookRegistered(priorityAndFairnessConfigConsumerHookName) {
 	} else if c.FlowControl != nil {
 		err := s.AddPostStartHook(priorityAndFairnessConfigConsumerHookName, func(hookContext PostStartHookContext) error {
-			go c.FlowControl.Run(hookContext.Done())
+			if err := c.FlowControl.Start(hookContext); err != nil {
+				klog.ErrorS(err, "failed starting flow control")
```

Comment:
```suggestion
				klog.FromContext(hookContext).Error(err, "Failed starting flow control")
```

`PostStartHookContext` implements `context.Context` and thus can be used to look up the logger.

Reply:
done